plugins/python-build/bin/python-build

```diff
@@ -1516,6 +1517,60 @@ use_xcode_sdk_zlib() {
   fi
 }
 
+use_homebrew_tcltk() {
+  # get the version from the folder that homebrew versions
+  local tcltk_version_long="$(ls "$(brew --cellar tcl-tk)")"
+  local tcltk_version="${tcltk_version_long%.*}"
+  local tcltk_flags="--with-tcltk-includes=-I$tcltk_libdir/include --with-tcltk-libs=-L$tcltk_libdir/lib -ltcl$tcltk_version -ltk$tcltk_version"
+  echo "python-build: use tcl-tk from homebrew"
+  package_option python configure --with-tcltk-libs="-L$tcltk_libdir/lib -ltcl$tcltk_version -ltk$tcltk_version"
+  package_option python configure --with-tcltk-includes="-I$tcltk_libdir/include"
+}
+
+use_custom_tcltk() {
+  # parse config options into array
+  IFS=$'\n'
+  local tcltk_libs_args=( $(xargs -n1 <<<"$PYTHON_CONFIGURE_OPTS") )
+
+  # get tcltk flag and options
+  local tcltk_ops_flag="--with-tcltk-libs="
+  local flag_index=0
+  local flag="${tcltk_libs_args[$flag_index]}"
+  until [[ "$flag" =~ "$tcltk_ops_flag" ]]; do
+    let flag_index++
+    flag="${tcltk_libs_args[$flag_index]}"
+  done
+
+  # get tcltk libs
+  local tcltk_libs="${flag//$tcltk_ops_flag/}"
+  # remove tcltk-flag from configure_opts
+  # this allows for weird input such as
+  # --with-tcltk-libs='    -L/custom-tcl-tk/lib -ltcl8.6 -ltk8.4  '
+  PYTHON_CONFIGURE_OPTS="${PYTHON_CONFIGURE_OPTS//"$tcltk_ops_flag"/}"
+  PYTHON_CONFIGURE_OPTS="${PYTHON_CONFIGURE_OPTS//$tcltk_libs/}"
+
+  # remove quotes, because there mess up compilations
+  PYTHON_CONFIGURE_OPTS="${PYTHON_CONFIGURE_OPTS//"''"/}"
+  PYTHON_CONFIGURE_OPTS="${PYTHON_CONFIGURE_OPTS//'""'/}"
+
+  echo "python-build: use tcl-tk from \$PYTHON_CONFIGURE_OPTS"
+  # echo "PYTHON_CONFIGURE_OPTS=${PYTHON_CONFIGURE_OPTS}"
+  package_option python configure --with-tcltk-libs="${tcltk_libs}"
+  IFS="$OLDIFS"
+}
```

Comment:
Why is this extraction-and-readding needed again?  We only need to detect if the user provided a custom value -- if they did, we don't do anything, if they didn't, we add a default.

Reply:
See #1125. 
Gist: The way python-build works write now is that it somehow does not pass multipart options correctly to python configure.
But I guess you are correct that a better way would be to fix the parsing of PYTHON_CONFIGURE_OPTS for any multipart options (not just tcl-tk).

Reply:
@native-api: How should we proceed about this? From my point of view fixing the general issue of #1125 is out of scope of this PR, however it solves 2 things:
(1) provide a working (e.g. homebrew) default for tcl-tk libs 
(2) improve the work-around for the problem described in #1125 by reliably parsing `PYTHON_CONFIGURE_OPTS` using `xargs` explicitly for tcl-tk. The old workaround for passing multipart tcl-tk to `python-build` used regex and was merge in #1397.

Since tcl-tk from homebrew is a long requested feature, I would argue in favour of merging this PR and opening an additional PR for refactoring the parsing of `PYTHON_CONFIGURE_OPTS` using `xargs`. I can set up a new draft PR for it, summarising the relevant information there.

Reply:
> From my point of view fixing the general issue of #1125 is out of scope of this PR

Okay, fair enough

Reply:
Let's settle on this then, shall we?

* Make the detection reliable (current regex is not) and split the detection part and reinsertion part
* Add a comment to the extract-and-reinsert code that it's a workaround for #1125 and should be removed once it's fixed.
* Resolve the conflicts with master

Then we can merge this.

Reply:
Just for clarification:

> Make the detection reliable (current regex is not) and split the detection part and reinsertion part

What do you mean by that?
Currently, this just checks if the tcl-tk-flag `"--with-tcltk-libs="` is a substring of `$PYTHON_CONFIGURE_OPTS`
```
if [[ "$PYTHON_CONFIGURE_OPTS" =~ $tcltk_ops_flag ]]; then
...assume the user has set sensible values for tcl-tk-libs
```

Should I rewrite it to use `*`wildcards? `if [[ "$PYTHON_CONFIGURE_OPTS" == *"$tcltk_ops_flag"* ]];`

Reply:
> What do you mean by that?

Detect if any of the args that `$PYTHON_CONFIGURE_OPTS` would be split into (honoring quotes) would be `--with-tcltk-libs=<whatever>`. The regex only detects if any of them contains this substring anywhere.

I admit that the chance of this substring randomly occurring elsewhere is pretty low...